### PR TITLE
[SQL-DS-CACHE-209][POAE7-1324] Integrate ape-common.jar and parquet_jni.so to ape-spark.jar to make …

### PR DESCRIPTION
**Pull request by ZJie1**
…it easier to use

## What changes were proposed in this pull request?

1. Combine the ape-common.jar with ape-spark.jar, so that we can just use one ape-spark.jar to run tests.
2. Convert the dynamic link library of libparquet_jni.so into a static link library to make it easier to distribute and use

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
 Pass the TPCH tests.

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by jikunshang** [comment]
Seems CI failed, can you check and update github action related files? 
CC @iyupeng

File: oap-ape/ape-java/ape-spark/pom.xml [hunk 1]
```diff
@@ -50,136 +51,204 @@
...
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+
+    <profile>
+      <id>native</id>
```
Comment (iyupeng) [inline oap-ape/ape-java/ape-spark/pom.xml]: `native` should be built before `shading`. Is there anything that guarantees this?

File: oap-ape/README.md [hunk 2]
```diff
@@ -69,14 +69,17 @@ cd IntelCodecLibrary/;
 mvn clean install
 ```
 
-Java build, we will use `$OAP_ROOT_DIR/oap-ape/ape-java/ape-common/target/ape-common-1.1.0-SNAPSHOT.jar`, `$OAP_ROOT_DIR/oap-ape/ape-java/ape-spark/target/ape-spark-1.1.0-SNAPSHOT.jar`, `$OAP_ROOT_DIR/oap-ape/ape-java/ape-flink/target/ape-flink-1.1.0-SNAPSHOT.jar` later.
+Both Java and native build, we will use `$OAP_ROOT_DIR/oap-ape/ape-java/ape-spark/target/ape-spark-1.1.0-SNAPSHOT.jar`, `$OAP_ROOT_DIR/oap-ape/ape-java/ape-flink/target/ape-flink-1.1.0-SNAPSHOT.jar` later.
```
Comment (iyupeng) [inline oap-ape/README.md]:
```suggestion
Native and Java build. We will use `$OAP_ROOT_DIR/oap-ape/ape-java/ape-spark/target/ape-spark-1.1.0-SNAPSHOT.jar`, `$OAP_ROOT_DIR/oap-ape/ape-java/ape-flink/ape-flink-x.xx.x/target/ape-flink-x.xx.x-1.1.0-SNAPSHOT.jar` later.
```

File: oap-ape/ape-java/ape-spark/pom.xml (hunk 1 above)
Comment (ZJie1) [reply oap-ape/ape-java/ape-spark/pom.xml]: Move native  compile into shading profile and  put it before package jar

File: oap-ape/ape-native/src/test/CMakeLists.txt [hunk 3]
```diff
@@ -43,31 +43,114 @@ add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/googletest-src
...
+    -Wl,-Bdynamic
+    -Wl,--as-needed
+    Threads::Threads
+    -Wl,--no-as-needed)
 add_test(NAME decimalTest COMMAND ${EXECUTABLE_OUTPUT_PATH}/decimalTest)
 
 add_executable(aggParserTest aggParserTest.cc)
-target_link_libraries(aggParserTest gtest_main parquet_jni)
+target_link_libraries(aggParserTest gtest_main parquet_jni
+   -Wl,-Bstatic
```
Comment (jikunshang) [inline oap-ape/ape-native/src/test/CMakeLists.txt]: could these parameters become one(or more) cmake variable to reduce repeat code?

Comment (ZJie1) [reply oap-ape/ape-native/src/test/CMakeLists.txt]: yes, changed them.

File: oap-ape/ape-native/src/CMakeLists.txt [hunk 4]
```diff
@@ -75,5 +75,37 @@ add_compile_options("-I$ENV{JAVA_HOME}/include/linux")
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -O3")
 
-target_link_libraries(parquet_jni ${ARROW_LIB} ${PARQUET_LIB} ${PLASMA_LIB} ${JNI} nlohmann_json::nlohmann_json OpenSSL::SSL ${HIREDIS_LIB} ${REDIS_PLUS_PLUS_LIB})
-target_link_libraries(parse nlohmann_json::nlohmann_json ${ARROW_LIB})
+INSTALL(TARGETS parquet_jni LIBRARY DESTINATION lib)
+
+LINK_DIRECTORIES(/lib /lib64 /usr/local/lib /usr )
```
Comment (jikunshang) [inline oap-ape/ape-native/src/CMakeLists.txt]: why need `/usr`

File: oap-ape/ape-native/src/CMakeLists.txt [hunk 5]
```diff
@@ -75,5 +75,37 @@ add_compile_options("-I$ENV{JAVA_HOME}/include/linux")
...
+
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+find_package(Arrow REQUIRED)
```
Comment (jikunshang) [inline oap-ape/ape-native/src/CMakeLists.txt]: move to the very beginning? and we did find arrow in L24

File: oap-ape/ape-native/src/CMakeLists.txt [hunk 6]
```diff
@@ -75,5 +75,37 @@ add_compile_options("-I$ENV{JAVA_HOME}/include/linux")
...
+
+target_link_libraries(parquet_jni
+    -Wl,-Bstatic
```
Comment (jikunshang) [inline oap-ape/ape-native/src/CMakeLists.txt]: refactor repeat parameters

File: oap-ape/ape-native/src/compile.sh [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+#
+
+set -eu
+# detect OS
+OS="`uname -s`"
+case ${OS} in
+  'Linux' )
+    OS='linux'
+    ;;
+  'Darwin')
```
Comment (jikunshang) [inline oap-ape/ape-native/src/compile.sh]: emmm actually I think MacOS is not supported as well.

File: oap-ape/ape-native/src/compile.sh [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+mkdir build
+pushd build
+cmake -DCMAKE_INSTALL_PREFIX=${RESOURCES_DIR} ../
+make
+make install
+popd
+
+popd
+set +eu
```
Comment (jikunshang) [inline oap-ape/ape-native/src/compile.sh]: add an empty line and add some log to indicate native build is completed successfully?

File: oap-ape/ape-native/src/test/CMakeLists.txt [hunk 9]
```diff
@@ -43,31 +43,39 @@ add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/googletest-src
 
 set(CMAKE_CXX_STANDARD 17)
 
+LINK_DIRECTORIES(/lib /lib64 /usr/local/lib /usr )
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+find_package(Arrow REQUIRED)
+
+set(ARROW_STATIC -Wl,-Bstatic; -Wl,--whole-archive; arrow_static; -Wl,--no-whole-archive)
+set(THREAD -Wl,-Bdynamic;-Wl,--as-needed;Threads::Threads;-Wl,--no-as-needed)
```
Comment (jikunshang) [inline oap-ape/ape-native/src/test/CMakeLists.txt:51]: just curious, why there are both `--as-needed` and `--no-as-needed` ? any reference?

File: oap-ape/ape-native/src/test/CMakeLists.txt [hunk 10]
```diff
@@ -43,31 +43,39 @@ add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/googletest-src
 
 set(CMAKE_CXX_STANDARD 17)
 
+LINK_DIRECTORIES(/lib /lib64 /usr/local/lib /usr )
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+find_package(Arrow REQUIRED)
```
Comment (jikunshang) [inline oap-ape/ape-native/src/test/CMakeLists.txt]: same as above.

File: oap-ape/ape-native/src/test/CMakeLists.txt (hunk 9 above)
Comment (ZJie1) [reply oap-ape/ape-native/src/test/CMakeLists.txt:51]: https://linux.die.net/man/1/ld

File: oap-ape/ape-native/src/CMakeLists.txt (hunk 4 above)
Comment (ZJie1) [reply oap-ape/ape-native/src/CMakeLists.txt]: find some dependency under /usr/include  /usr/share  /usr/bin. So added it.

File: oap-ape/ape-native/src/compile.sh (hunk 7 above)
Comment (ZJie1) [reply oap-ape/ape-native/src/compile.sh]: remove it.

File: oap-ape/ape-native/src/CMakeLists.txt (hunk 4 above)
Comment (jikunshang) [reply oap-ape/ape-native/src/CMakeLists.txt]: seems `LINK_DIRECTORIES` is not recommanded, see https://cmake.org/cmake/help/latest/command/link_directories.html

**Comment by jikunshang** [comment]
Overall LGTM.
Please take a review.  @winningsix @yma11

File: oap-ape/README.md [hunk 11]
```diff
@@ -88,11 +90,11 @@ sudo cp ./lib/libparquet_jni.so /usr/lib/
 
 ## Run Spark with APE
  1. make sure you have installed `libparquet_jni.so` on all of your worker nodes.
```
Comment (yma11) [inline oap-ape/README.md]: still need manually install libparquet_jni.so as it's bundled with ape-spark.jar in this PR?

Comment (ZJie1) [reply oap-ape/README.md]: modify this. APE server still need manually install it.

**Comment by yma11** [comment]
+1.

**Pull request merged**